tests/wpt/servowpt.py

```diff
@@ -186,41 +188,106 @@ def update_tests(**kwargs):
+        repo_url = f"https://github.com/{repository}"
 
-class GitHubQueryFilter():
-    def __init__(self, token):
-        self.token = token
+        run_id = github_context['run_id']
+        build_url = f"{repo_url}/actions/runs/{run_id})"
 
-    def is_failure_intermittent(self, test_name):
-        url = "https://api.github.com/search/issues?q="
-        query = "repo:servo/servo+" + \
-            "label:I-intermittent+" + \
-            "type:issue+" + \
-            "state:open+" + \
-            test_name
+        commit_title = github_context["event"]["head_commit"]["message"]
+        match = re.match(r"^Auto merge of #(\d+)", commit_title)
+        pr_url = f"{repo_url}/pull/{match.group(1)}" if match else ""
 
-        # we want `/` to get quoted, but not `+` (github's API doesn't like
-        # that), so we set `safe` to `+`
-        url += urllib.parse.quote(query, safe="+")
+        return GithubContextInformation(
+            build_url,
+            pr_url,
+            github_context["ref_name"]
+        )
 
-        request = urllib.request.Request(url)
-        request.add_header("Authorization", f"Bearer: {self.token}")
-        request.add_header("Accept", "application/vnd.github+json")
-        return json.load(
-            urllib.request.urlopen(request)
-        )["total_count"] > 0
+    def make_data_from_result(
+        self,
+        result: Union[UnexpectedResult, UnexpectedSubtestResult],
+    ) -> dict:
+        data = {
+            'path': result.path,
+            'subtest': None,
+            'expected': result.expected,
+            'actual': result.actual,
+            'time': result.time // 1000,
+            'message': result.message,
+            'stack': result.stack,
+        }
+        if isinstance(result, UnexpectedSubtestResult):
+            data["subtest"] = result.subtest
+        return data
+
+    def filter_unexpected_results(
+        self,
+        unexpected_results: List[UnexpectedResult]
+    ) -> Tuple[List[UnexpectedResult], List[UnexpectedResult]]:
+        attempts = []
+        for result in unexpected_results:
+            attempts.append(self.make_data_from_result(result))
+            for subtest_result in result.unexpected_subtest_results:
+                attempts.append(self.make_data_from_result(subtest_result))
+
+        context = self.get_github_context_information()
+        try:
+            request = urllib.request.Request(
+                url=self.url,
+                method='POST',
+                data=json.dumps({
+                    'branch': context.branch_name,
+                    'build_url': context.build_url,
+                    'pull_url': context.pull_url,
+                    'attempts': attempts
+                }).encode('utf-8'),
+                headers=self.headers)
+
+            known_intermittents = dict()
+            with urllib.request.urlopen(request) as response:
+                for test in json.loads(response.read())["known"]:
```

Comment:
nit: i think this can be simplified to

```python
for test in json.load(response)["known"]:
```

Reply:
Nice catch. I've fixed this in the latest version.